source/dofs/dof_tools.inst.in

```diff
@@ -821,5 +803,27 @@ for (deal_II_dimension : DIMENSIONS; deal_II_space_dimension : DIMENSIONS)
         get_dofs_on_patch<hp::DoFHandler<deal_II_dimension,deal_II_space_dimension> >
         (const std::vector<hp::DoFHandler<deal_II_dimension,deal_II_space_dimension>::active_cell_iterator> &patch);
       \}
+
+
+template void DoFTools::extract_dofs<deal_II_dimension,deal_II_space_dimension>
+(const DoFHandler<deal_II_dimension,deal_II_space_dimension>&,
+ const ComponentMask &,
+ std::vector<bool>&);
```

Comment:
We instantiate DoFHandler only for dim <= spacedim, so I would suggest to put these instantiations inside the count_dofs_on_patch conditional (it might result in undefined references otherwise). I would also suggest to follow the style of those functions (i.e., open the namespace DoFTools rather than saying DoFTools::extract_dofs). It's not wrong the way you do it, but more consistent with the style we try to follow.